web/client-api/src/main/java/io/deephaven/web/client/api/subscription/AbstractTableSubscription.java

```diff
@@ -0,0 +1,507 @@
+        notifyUpdate(rowsAdded, rowsRemoved, totalMods, shifted);
+    }
+
+    protected void notifyUpdate(RangeSet rowsAdded, RangeSet rowsRemoved, RangeSet totalMods, ShiftedRange[] shifted) {
+        // TODO (deephaven-core#2435) Rewrite shifts as adds/removed/modifies
+        UpdateEventData detail = new SubscriptionEventData(
+                barrageSubscription,
+                rowStyleColumn,
+                columns,
+                transformRowsetForConsumer(rowsAdded, barrageSubscription.getServerViewport(),
+                        barrageSubscription.isReversed()),
+                transformRowsetForConsumer(rowsRemoved, barrageSubscription.getServerViewport(),
+                        barrageSubscription.isReversed()),
+                transformRowsetForConsumer(totalMods, barrageSubscription.getServerViewport(),
+                        barrageSubscription.isReversed()),
+                barrageSubscription.getServerViewport() != null ? null : shifted);
+        CustomEventInit<UpdateEventData> event = CustomEventInit.create();
+        event.setDetail(detail);
+        fireEvent(TableSubscription.EVENT_UPDATED, event);
+    }
+
+    public static class SubscriptionRow implements TableData.Row {
+        private final WebBarrageSubscription subscription;
+        private final int rowStyleColumn;
+        protected final long index;
+        public LongWrapper indexCached;
+
+        public SubscriptionRow(WebBarrageSubscription subscription, int rowStyleColumn, long index) {
+            this.subscription = subscription;
+            this.rowStyleColumn = rowStyleColumn;
+            this.index = index;
+        }
+
+        @Override
+        public LongWrapper getIndex() {
+            if (indexCached == null) {
+                indexCached = LongWrapper.of(index);
+            }
+            return indexCached;
+        }
+
+        @Override
+        public Any get(Column column) {
+            return subscription.getData(index, column.getIndex());
+        }
+
+        @Override
+        public Format getFormat(Column column) {
+            long cellColors = 0;
+            long rowColors = 0;
+            String numberFormat = null;
+            String formatString = null;
+            if (column.getStyleColumnIndex() != null) {
+                LongWrapper wrapper = subscription.getData(index, column.getStyleColumnIndex()).uncheckedCast();
+                cellColors = wrapper == null ? 0 : wrapper.getWrapped();
+            }
+            if (rowStyleColumn != TableData.NO_ROW_FORMAT_COLUMN) {
+                LongWrapper wrapper = subscription.getData(index, rowStyleColumn).uncheckedCast();
+                rowColors = wrapper == null ? 0 : wrapper.getWrapped();
+            }
+            if (column.getFormatStringColumnIndex() != null) {
+                numberFormat = subscription.getData(index, column.getFormatStringColumnIndex()).uncheckedCast();
+            }
```

Comment:
I think `numberFormat` can be removed as well, it's been removed in favour of `formatString`.

In any case, this if block and the next if block are both `column.getFormatStringColumnIndex()` which is incorrect; this if block was `column.getFormatColumnIndex()` which was deprecated and we removed, pretty sure we can remove this/`numberFormat` as well.

Commit where I deprecated `numberFormat`: https://github.com/deephaven-ent/iris/commit/a1f272f783f900ac136b15b5f4beabea78b8d8ae

Reply:
Excellent, I wasn't sure that it had arrived at the "we can safely remove this" phase.